third_party/nvfuser/test/test_resize.cpp

```diff
@@ -0,0 +1,1594 @@
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, aten_inputs);
+  auto cg_outputs = fe.runFusion(aten_inputs);
+
+  auto ref = at::cat({t0, t1}, 1);
+
+  TORCH_CHECK(ref.equal(cg_outputs[0]));
+}
+
+// Cat + arith op
+TEST_F(NVFuserTest, FusionResizeCat5_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+  auto tv1 = makeSymbolicTensor(2);
+  fusion.addInput(tv1);
+  auto tv2 = makeSymbolicTensor(2);
+  fusion.addInput(tv2);
+
+  auto tv3 = cat({tv0, tv1}, 1);
+  auto tv4 = add(tv3, tv2);
+  fusion.addOutput(tv4);
+
+  tv4->merge(0);
+  tv4->split(0, 128);
+
+  TransformPropagator propagator(tv4);
+  MaxRootDomainInfoSpanningTree(tv4).traverse(&propagator);
+
+  inlineMost();
+
+  tv4->axis(0)->parallelize(ParallelType::BIDx);
+  tv4->axis(1)->parallelize(ParallelType::TIDx);
+  scheduler_utils::parallelizeAllLike(tv4);
+
+  std::vector<int64_t> shape0({11, 12});
+  std::vector<int64_t> shape1({shape0[0], 13});
+  std::vector<int64_t> shape2({shape0[0], shape0[1] + shape1[1]});
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::manual_seed(0);
+
+  auto t0 = at::randn(shape0, options);
+  auto t1 = at::randn(shape1, options);
+  auto t2 = at::randn(shape2, options);
+  std::vector<c10::IValue> aten_inputs({t0, t1, t2});
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, aten_inputs);
+  auto cg_outputs = fe.runFusion(aten_inputs);
+
+  auto ref = at::cat({t0, t1}, 1) + t2;
+
+  testValidate(&fusion, cg_outputs, aten_inputs, {ref}, __LINE__, __FILE__);
+}
+
+// Cat 3 tensors
+TEST_F(NVFuserTest, FusionResizeCat6_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  std::vector<int64_t> shape0({2, 4});
+  std::vector<int64_t> shape1({5, 4});
+  std::vector<int64_t> shape2({3, 4});
+
+  auto tv0 = makeConcreteTensor(shape0);
+  fusion.addInput(tv0);
+  auto tv1 = makeConcreteTensor(shape1);
+  fusion.addInput(tv1);
+  auto tv2 = makeConcreteTensor(shape2);
+  fusion.addInput(tv2);
+
+  auto tv3 = cat({tv0, tv1, tv2}, 0);
+  fusion.addOutput(tv3);
+
+  tv3->merge(0);
+  tv3->split(0, 4);
+  TransformPropagator propagator(tv3);
+  MaxRootDomainInfoSpanningTree(tv3).traverse(&propagator);
+
+  inlineMost();
+
+  tv3->axis(0)->parallelize(ParallelType::BIDx);
+  tv3->axis(1)->parallelize(ParallelType::TIDx);
+  scheduler_utils::parallelizeAllLike(tv3);
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::manual_seed(0);
+
+  auto t0 = at::randn(shape0, options);
+  auto t1 = at::randn(shape1, options);
+  auto t2 = at::randn(shape2, options);
+  std::vector<c10::IValue> aten_inputs({t0, t1, t2});
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion, aten_inputs);
+  auto cg_outputs = fe.runFusion(aten_inputs);
+
+  auto ref = at::cat({t0, t1, t2}, 0);
+
+  TORCH_CHECK(ref.equal(cg_outputs[0]));
+}
+
+// Cat many tensors
+TEST_F(NVFuserTest, FusionResizeCat7_CUDA) {
```

Comment:
Do you check that an error gets thrown if runtime values don't support the requested concat?

Reply:
The type and number of dimensions must match:
 https://github.com/csarofeen/pytorch/pull/2480/files/82845fa11d79a153f62fd11fa4daf473476cd728#diff-bee9b138517ab66ac5e215f0603aacbacc8984fda748a6d636e36595ec0b12deR467-R494.

But that's it. For example, the extents of the non-concatenated dimensions must match, but that's not checked, like normal arithmetic ops where we assume two operands have the same shape.

Reply:
I think it makes sense to add a runtime check for this.

Reply:
I'm not sure what to check. We don't check, for example, in `add(tv0, tv1)`, `tv0` and `tv1` have the same shape, but rather we assume that's the case and use that to build the IterDomain graph. Following the same approach, non-concatenated domains should be assumed to be the same.

Reply:
Heh, that's a good point, and that's why this would error out during expression evaluator as our evaluated symbols for parallelization wouldn't match. Okay, will put it in the large bucket of things we should do. Would be nice to have a validation check that everything matches as expected across the iter domain relationships.

Reply:
Something like a pedantic mode, where everything is validated no matter how high its overhead would be.